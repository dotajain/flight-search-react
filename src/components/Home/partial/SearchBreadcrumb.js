import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import moment from 'moment';

import { connect } from 'react-redux';

const SearchBreadcrumb = ({ searchRequest }) => {
  const origin = _.split(searchRequest.origin, ' - ')[1];
  const destination = _.split(searchRequest.destination, ' - ')[1];

  let returnCity;
  let returnDate;

  if (searchRequest.returnDate) {
    returnCity = <span>{origin}</span>;
    returnDate = (
      <div className="search__dates--item">
        <span className="search__dates--label">Return:</span>{' '}
        <span className="search__dates--value">
          {moment(searchRequest.returnDate).format('DD-MMM-YYYY')}
        </span>
      </div>
    );
  }

  return (
    <div className="search__result--header">
      <div className="breadcrumbs">
        <span>{origin}</span>
        <span>{destination}</span>
        {returnCity}
      </div>
      <div className="box search__dates">
        <div className="search__dates--item">
          <span className="search__dates--label">Depart:</span>{' '}
          <span className="search__dates--value">
            {moment(searchRequest.departureDate).format('DD-MMM-YYYY')}
          </span>
        </div>
        {returnDate}
      </div>
    </div>
  );
};

SearchBreadcrumb.propTypes = {
  searchRequest: PropTypes.object,
};

const mapStateToProps = state => ({
  searchRequest: state.flightSearch.searchRequest,
});
export default connect(mapStateToProps)(SearchBreadcrumb);
