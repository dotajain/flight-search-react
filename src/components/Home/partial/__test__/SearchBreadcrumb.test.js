import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../store';

import SearchBreadcrumb from '../SearchBreadcrumb';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered SearchBreadcrumb Component', () => {
  const component = renderer.create(<Provider store={store}><SearchBreadcrumb /></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
