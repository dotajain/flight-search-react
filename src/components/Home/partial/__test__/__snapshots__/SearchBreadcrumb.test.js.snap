// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered SearchBreadcrumb Component 1`] = `
<div
  className="search__result--header"
>
  <div
    className="breadcrumbs"
  >
    <span />
    <span />
  </div>
  <div
    className="box search__dates"
  >
    <div
      className="search__dates--item"
    >
      <span
        className="search__dates--label"
      >
        Depart:
      </span>
       
      <span
        className="search__dates--value"
      >
        16-Oct-2017
      </span>
    </div>
  </div>
</div>
`;
